### PR TITLE
Fixes #3636 Code analysis not updated after changing packages

### LINKED ISSUE #3636: Code analysis features not updated after package is installed or removed
New Bottle Web Project
Accept Virtual Env + Install requirements option
app.py automatically opens

Notice the squiggles on `import bottle`
Wait for virtual env + package install to finish

Result: Squiggles don't go away. No completions on `bottle.`.

Close/reopen file doesn't help. You have to close/reopen solution.

The reverse problem also exists. Now that you've reloaded the solution and get completions on `bottle`, uninstall the package.

Result: Squiggles don't appear. Close/reopen file doesn't help. You have to close/reopen solution.

### PR DESCRIPTION
**Pull request by zooba**
Fixes #3636 Code analysis not updated after changing packages
Adds event handling for changed packages.
Adds ObjectDisposedException handling for timers.

### CODE REVIEW

File: Python/Product/PythonTools/PythonTools/Project/PythonProjectNode.cs [hunk 1]
```diff
@@ -233,6 +249,12 @@ internal class PythonProjectNode :
                     }
                 }
 
+                _activePackageManagers = InterpreterOptions.GetPackageManagers(_active).ToArray();
```
Comment (AlexanderSher) [inline Python/Product/PythonTools/PythonTools/Project/PythonProjectNode.cs]: `_active` can be `null`

Comment (zooba) [reply Python/Product/PythonTools/PythonTools/Project/PythonProjectNode.cs]: Now it does.

Comment (AlexanderSher) [reply Python/Product/PythonTools/PythonTools/Project/PythonProjectNode.cs]: Alright, let me test it

File: Python/Product/Analysis/Interpreter/Ast/AstPythonInterpreterFactory.cs [hunk 2]
```diff
@@ -125,15 +125,7 @@ InterpreterFactoryCreationOptions options
...
-                for (int retries = 5; retries > 0; --retries) {
-                    try {
-                        File.Delete(_searchPathCachePath);
-                        break;
-                    } catch (IOException) {
-                    } catch (UnauthorizedAccessException) {
-                    }
-                    Thread.Sleep(10);
-                }
+                PathUtils.DeleteFile(_searchPathCachePath);
```
Comment (AlexanderSher) [inline Python/Product/Analysis/Interpreter/Ast/AstPythonInterpreterFactory.cs]: Don't you call `File.Exists` in `PathUtils.DeleteFile`?

Comment (zooba) [reply Python/Product/Analysis/Interpreter/Ast/AstPythonInterpreterFactory.cs]: No, it will retry and give up without an error, but it doesn't check.

**Pull request merged**